### ISSUE TITLE
Module settings get stuck under browser menu

### ISSUE THREAD

**Issue by websbyjd** [elegantthemes/Divi-Beta#605]
### Problem:

I've been doing a Divi project and I'm noticing that in the VB I keep getting the module settings box stuck up under the browser menu.  Once this happens every settings box opened is in the same position.  The only way to reset it is to close VB and reopen.   Not a huge issue, but annoying nonetheless.  I attached a screenshot link

### Steps To Reproduce:
1.  Open VB
2. open a module setting box
3. drag it up to the top of your viewport and release it
4. try to go back and move it again

### Additional Information:

*Please include any of the following details that are relevant to the issue you are reporting.*

**Device**: PC

**Operating System**: Win10

**Browser**: Chrome

**Screen Resolution**: 1366 x 768

**Screenshot or Screencast**: [http://youtu.be/qMx6mAf_G3Y?hd=1]

**Link To Download Layout JSON File**:

-

#### Before submitting your issue, please ensure the following:

* You searched this Github repo for the issue and found no open reports of the issue.
* You properly formated the issue title - eg. `Scope :: One sentence to describe the issue`.
* You have provided a detailed description of the issue.
* You have provided any/all relevant information: screenshot, screencast, page URL.
* You have provided the steps that one must take to reproduce the issue.

[View Example](https://github.com/elegantthemes/Divi-Beta/tree/master/.github/ISSUE_EXAMPLE.md)

**Comment by timayhelen** [comment]
Thanks for the report. For the time being you can press Shift + Tab if you ever get stuck like this again. It will snap the modal to the side of the page and you can drag it back off again.

**Comment by fikrirasyid** [comment]
@websbyjd thank you for reporting the issue. A patch for this issue have been submitted and currently under review process. Once it passes the review process, it should be included on the Divi release.

Meanwhile, here's how it looks:

![modal s max top position](https://cloud.githubusercontent.com/assets/916442/19277516/20d3a486-9004-11e6-8a87-7b60e2fa7fef.gif)

**Comment by lots0logs** [comment]
@fikrirasyid Is there any way we can prevent the browser from selecting all of the page's text content when the cursor goes out-of-bounds while dragging the modal? Also, I think we should do the same for the left and right boundaries..and the bottom as well (not until the page can't be scrolled anymore of course).

**Comment by timayhelen** [comment]
@lots0logs @fikrirasyid It will get annoying if the modal is restricted on left/right/bottom. Just imagine for a moment if this was how windows worked in your OS. It would be super frustrating. 

Let's keep things how they are in this PR. It works exactly the way you would expect it to, given normal window conventions. 

![window](https://cloud.githubusercontent.com/assets/3333028/19306158/2f20cf08-9029-11e6-9a24-3551c83fa597.gif)

**Comment by fikrirasyid** [comment]
@timayhelen noted. I'm keeping it the way thing is right now then.

**Comment by fikrirasyid** [comment]
@lots0logs I've pushed a fix which disable text selection during modal drag / resize to avoid unwanted text selection at https://github.com/elegantthemes/submodule-builder/commit/63a3be47d1a3dc016fd3c8a9bb730ee0acae5a6a

**Comment by timayhelen** [comment]
@fikrirasyid Can we remove this class from being added/removed during the Modal Snap animation? I don't think it's needed now that it always exists during modal move.

**Comment by fikrirasyid** [comment]
@timayhelen sure, I've reverted the commit so there is no additional classes added when settings modal is dragged / resized.

**Comment by timayhelen** [comment]
@fikrirasyid I meant we should keep the class being added when you drag and resize the modal, but it's no longer needed when the "modal snap" placeholder animation is triggered.

Basically you already implemented this fix for when the modal overflowed the left/right viewport. But if the class exists all the time while dragging, we don't need to add and remove it when the modal snap placeholder appears.

**Comment by timayhelen** [comment]
![snap](https://cloud.githubusercontent.com/assets/3333028/19317885/5cbcc494-905b-11e6-81f0-ba3481f73eb7.gif)

**Comment by timayhelen** [comment]
@fikrirasyid I went ahead and made the change myself, but then noticed that when the body class is toggled on and off, it really glitches up the react-motion animation for modal drag and resize. So I don't think it's worth adding. Let's just leave this PR how it is.

**Comment by fikrirasyid** [comment]
I investigated this yesterday before I signed off and hadn't found a decent solution for it. Basically snap happens during drag start and drag end which causes the issue. 

I'll open another issue for the unwanted text selection bug so this can be merged.

(aand this PR is merged as I write this 😅)
